### PR TITLE
Add missing super __init__ in subclasses

### PR DESCRIPTION
**Pull request by QuLogic**
## PR Summary

These can cause issues for some static checks (I noticed them in LGTM), and mostly the code in the super `__init__` is duplicated in the subclass.

## PR Checklist

- [x] Has pytest style unit tests (and `pytest` passes).
- [x] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [n/a] New features are documented, with examples if plot related.
- [n/a] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [x] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [n/a] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [n/a] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/textpath.py [hunk 1]
```diff
@@ -385,11 +385,11 @@ def __init__(self, xy, s, size=None, prop=None,
 
         self._cached_vertices = None
         s, ismath = Text(usetex=usetex)._preprocess_math(s)
-        self._vertices, self._codes = text_to_path.get_text_path(
-            prop, s, ismath=ismath)
+        super().__init__(
```
Comment (anntzer) [inline lib/matplotlib/textpath.py:388]: perhaps rather use fast_from_codes_and_verts for speed?

Comment (QuLogic) [reply lib/matplotlib/textpath.py:388]: That's a class method / constructor, though?

Comment (anntzer) [reply lib/matplotlib/textpath.py:388]: ah yes.  Oh well, I guess this isn't a bottleneck...

**Review by anntzer — APPROVED** [review]

File: lib/matplotlib/mathtext.py [hunk 2]
```diff
@@ -189,6 +189,7 @@ class MathtextBackendPs(MathtextBackend):
         "_PSResult", "width height depth pswriter used_characters")
 
     def __init__(self):
+        super().__init__()
```
Comment (jklymak) [inline lib/matplotlib/mathtext.py:192]: Are none of these really ever tested?  Maybe beyond this PR, but they should be if they aren't...

Comment (QuLogic) [reply lib/matplotlib/mathtext.py:192]: They're almost all unused, which I guess is why they're deprecated. For the ones that _are_ used, I assume they all get `.set_canvas_size` called first, which sets the things that were missed in the super `__init__`.

**Review by tacaswell — APPROVED** [review]

**Pull request merged**